os/kernel/task/task_create.c

```diff
@@ -194,10 +194,7 @@ static int thread_create(FAR const char *name, uint8_t ttype, int priority, int
 	}
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
 	/* Update the pid information in stack node */
-	struct mm_allocnode_s *node;
-
-	node = (struct mm_allocnode_s *)(tcb->cmn.stack_alloc_ptr - SIZEOF_MM_ALLOCNODE);
-	node->pid = (-1) * (tcb->cmn.pid);
+	heapinfo_update_stack_pid(tcb->cmn.stack_alloc_ptr, tcb->cmn.pid)
```

Comment:
Can you please check the case of vfork() also?
Currently we are not setting the negative pid when stack is created in vfork API. But i think we need to do it.

Reply:
added, thanks~